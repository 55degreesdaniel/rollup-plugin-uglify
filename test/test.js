import test from 'ava';
import { rollup } from 'rollup';
import { minify } from 'uglify-js';
import uglify from '..';
import { readFileSync as readFile } from 'fs';

test('should minify', t => {
    return rollup({
        entry: 'fixtures/unminified.js',
        plugins: [ uglify() ]
    }).then(bundle => {
        const unminified = readFile('fixtures/unminified.js', 'utf-8');
        const result = bundle.generate({
            format: 'cjs'
        });
        t.is(result.code, minify(unminified, { fromString: true }).code);
    });
});

test('should minify via uglify options', t => {
    return rollup({
        entry: 'fixtures/empty.js',
        plugins: [ uglify({
            output: { comments: 'all' }
        }) ]
    }).then(bundle => {
        const result = bundle.generate({
            banner: '/* package name */',
            format: 'cjs'
        });

        t.is(result.code, '/* package name */\n"use strict";');
    });
});

test('should minify with sourcemaps', t => {
    return rollup({
        entry: 'fixtures/sourcemap.js',
        plugins: [ uglify() ]
    }).then(bundle => {
        const result = bundle.generate({
            format: 'cjs',
            sourceMap: true
        });

        t.ok(result.map, 'has a source map');
        t.is(result.map.version, 3, 'source map has expected version');
        t.ok(Array.isArray(result.map.sources), 'source map has sources array');
        t.is(result.map.sources.length, 2, 'source map has two sources');
        t.ok(Array.isArray(result.map.names), 'source maps has names array');
        t.ok(result.map.mappings, 'source map has mappings');
    });
});

test('should allow passing minifier', t => {
    const expectedCode = readFile('fixtures/plain-file.js', 'utf-8');
    const testOptions = {
        foo: 'bar'
    };

    return rollup({
        entry: 'fixtures/plain-file.js',
        plugins: [ uglify(testOptions, (code, options) => {
            t.ok(code, 'has unminified code');
            t.is(`${code}\n`,
                expectedCode,
                'expected file content is passed to minifier');
            t.ok(options, 'has minifier options');
            t.is(options.foo, 'bar', 'minifier gets custom options');

            return { code };
        })]
    }).then(bundle => {
        const result = bundle.generate();

        t.ok(result.code, 'result has return code');
        t.is(`${result.code}\n`,
            expectedCode,
            'result code has expected content');
    });
});
