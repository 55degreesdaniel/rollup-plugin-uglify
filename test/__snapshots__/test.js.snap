// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works with code splitting 1`] = `
Object {
  "chunk-1.js": Object {
    "code": "var chunk1=\\"chunk-1\\";console.log(chunk1);
",
    "exports": Array [],
    "fileName": "chunk-1.js",
    "imports": Array [],
    "isEntry": true,
    "map": null,
    "modules": Object {
      "/Users/trysound/host/rollup-plugin-uglify/test/fixtures/chunk-1.js": Object {
        "originalLength": 44,
        "removedExports": Array [],
        "renderedExports": Array [],
        "renderedLength": 44,
      },
    },
  },
  "chunk-2.js": Object {
    "code": "var chunk2=\\"chunk-2\\";console.log(chunk2);
",
    "exports": Array [],
    "fileName": "chunk-2.js",
    "imports": Array [],
    "isEntry": true,
    "map": null,
    "modules": Object {
      "/Users/trysound/host/rollup-plugin-uglify/test/fixtures/chunk-2.js": Object {
        "originalLength": 44,
        "removedExports": Array [],
        "renderedExports": Array [],
        "renderedLength": 44,
      },
    },
  },
}
`;
